### PR TITLE
Switch to Mason

### PR DESCRIPTION
**Pull request by lukasmartinelli**
#18. Waiting on https://github.com/mapbox/mason/pull/462.

### CODE REVIEW

**Comment by lukasmartinelli** [comment]
@springmeyer Not urgent but do you think the rocksdb linking failure https://travis-ci.org/mapbox/osm-tag-history/jobs/262836070 does have to do with how we make the static rocksdb lib in https://github.com/mapbox/mason/pull/462.

```
(.text._ZN7rocksdb14BZip2_CompressERKNS_18CompressionOptionsEjPKcmPSs[_ZN7rocksdb14BZip2_CompressERKNS_18CompressionOptionsEjPKcmPSs]+0xc9): undefined reference to `BZ2_bzCompress'
../mason_packages/linux-x86_64/rocksdb/5.4.6/lib/librocksdb.a(block_based_table_builder.o): In function `rocksdb::BZip2_Compress(rocksdb::CompressionOptions const&, unsigned int, char const*, unsigned long, std::string*)':
(.text._ZN7rocksdb14BZip2_CompressERKNS_18CompressionOptionsEjPKcmPSs[_ZN7rocksdb14BZip2_CompressERKNS_18CompressionOptionsEjPKcmPSs]+0xf6): undefined reference to `BZ2_bzCompressEnd'
collect2: error: ld returned 1 exit status
make[2]: *** [add_tags] Error 1
make[1]: *** [CMakeFiles/add_tags.dir/all] Error 2
make: *** [all] Error 2
```

Because the OSX build on Travis works 😱 

https://travis-ci.org/mapbox/osm-tag-history/jobs/262836066

**Comment by springmeyer** [comment]
@lukasmartinelli - Can you try making the build verbose so we can see the compiler/linker output? https://stackoverflow.com/questions/41644373/why-does-cmake-verbose-1-give-verbose-cmake-output-but-cmake-dcmake-verbos

I think it will show that `-lbzip2` is coming before `-lrocksdb`. Why this won't work is described at https://github.com/mapbox/cpp/blob/master/glossary.md#linking-order. So the fix would be:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index c86d48e..5445fd4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,7 +46,7 @@ mason_use(rocksdb VERSION 5.4.6)
 include_directories(SYSTEM ${MASON_PACKAGE_rocksdb_INCLUDE_DIRS})
 set(ROCKSDB_LIBRARIES ${MASON_PACKAGE_rocksdb_STATIC_LIBS})
 
-set(ALL_LIBRARIES ${ZLIB_LIBRARY} ${EXPAT_LIBRARIES} ${BZIP2_LIBRARIES} ${ROCKSDB_LIBRARIES})
+set(ALL_LIBRARIES ${EXPAT_LIBRARIES} ${ROCKSDB_LIBRARIES} ${ZLIB_LIBRARY} ${BZIP2_LIBRARIES} )
 #----------------------------------------------------------------------
 add_executable(build_tag_lookup build_tag_lookup.cpp)
 add_executable(add_tags add_tags.cpp)
```

File: .travis.yml [hunk 1]
```diff
@@ -1,49 +1,40 @@
...
-        - os: osx
-          compiler: clang
-          env: BUILD_TYPE=Release
+  fast_finish: true
+  include:
+    - os: linux
+      addons:
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['g++-7', 'cmake', 'realpath', 'libbz2-dev', 'zlib1g-dev', 'libexpat1-dev', 'libboost-all-dev']
```
Comment (springmeyer) [inline .travis.yml:15]: You can drop `'libbz2-dev', 'zlib1g-dev', 'libexpat1-dev', 'libboost-all-dev'` since these are all coming from mason now.

Comment (lukasmartinelli) [reply .travis.yml:15]: Thanks @jenningsanderson for fixing in https://github.com/mapbox/osm-tag-history/commit/36adfbe27cf477d828a6fe9fb4599d1dc9b09f20

**Comment by lukasmartinelli** [comment]
> @lukasmartinelli - Can you try making the build verbose so we can see the compiler/linker output? https://stackoverflow.com/questions/41644373/why-does-cmake-verbose-1-give-verbose-cmake-output-but-cmake-dcmake-verbos

Thanks a lot Dane!!! ❤️ Really appreciate the help. 

Was able to fix it. Linux builds work now!
Learning a lot. Linking order matters!

Also had to add `pthreads` to the libraries linked.

**Comment by springmeyer** [comment]
👍  Glad you got things working @lukasmartinelli @jenningsanderson

**Pull request merged**